learn/develop/sparse-data/index.qmd

```diff
@@ -0,0 +1,92 @@
+
+These sparse vectors can be put into tibbles and are what allow tidymodels to handle sparse data. We will henceforth refer to a tibble that contains sparse columns as a **sparse tibble**. Sparse matrices require all the elements to be of the same type. All logical, all integer, or all doubles. This limitation does not apply to sparse tibbles as you can store both sparse vectors and dense vectors. But also mix and match numeric, factors, datetimes, and more.
+
+The sparsity mostly matters for the predictors. The outcomes, case weights, and predictions are done densely.
+
+Below is outlined how sparse data (matrix & tibble) works with the various tidymodels packages.
+
+## rsample
+
+The resampling functions from the rsample package work with sparse tibbles out of the box. However, they won't work with sparse matrices. Instead, use the `coerce_to_sparse_tibble()` function from the sparsevctrs package to turn the sparse matrix into a sparse tibble and proceed with that object.
+
+```r
+library(sparsevctrs)
+data_tbl <- coerce_to_sparse_tibble(data_mat)
+```
+
+## recipes
+
+The recipes package receives data at 3 different places, in `recipe()`, `prep()`, and `bake()`. These functions all handle sparse data in the same matter.
+
+Sparse tibbles should work as normal. Sparse matrices are accepted and then turned into sparse tibbles right away, and flow through the internals as sparse tibbles.
+
+The `composition` argument of `bake()` understands sparse tibbles. When `composition = "dgCMatrix"` the resulting sparse matrix will be created from the sparse tibble with minimal overhead.
+
+The different recipes steps don't know how to handle sparse vectors yet. This will mean that most likely it will result in the materialization of the selected variables. This is a known issue and is planned to be fixed.
+
+## parsnip
+
+The parsnip package receives data in 3 places, in `fit()`, `fit_xy()`, and `predict()`.
+
+Sparse data does not yet work with the formula interface for `fit()`. This is a known issue and is expected to be fixed long term.
+
+For the remaining interfaces for `fit()` and `fit_xy()`, both sparse tibbles and sparse matrices are supported. Sparse matrices are turned into sparse tibbles early on. When fitting a model in parsnip. It checks whether the engine supports sparse matrices using the `allow_sparse_x` specification. A warning is thrown if sparse data is passed to an engine that doesn't support it, informing the user of that fact and that the data will be converted to a dense representation.
```

Comment:
```suggestion
For the remaining interfaces for `fit()` and `fit_xy()`, both sparse tibbles and sparse matrices are supported. Sparse matrices are turned into sparse tibbles early on. When fitting a model in parsnip, the package checks whether the engine supports sparse matrices using the `allow_sparse_x` specification. A warning is thrown if sparse data is passed to an engine that doesn't support it, informing the user of that fact and that the data will be converted to a dense representation.
```

Reply:
re: this paragraph and the previous: parsnip has no `fit()` interface other than the formula interface. i.e. if you're using `fit()` rather than `fit_xy()`, you're using a formula. Maybe just start this with "For `fit_xy()`" and adjust the verbiage in the previous?